bot/exts/moderation/infraction/infractions.py

```diff
@@ -480,6 +474,45 @@ async def apply_ban(
         async def action() -> None:
             await ctx.guild.ban(user, reason=reason, delete_message_days=purge_days)
 
+        if isinstance(user, Member) and any(role.id in constants.STAFF_PARTNERS_COMMUNITY_ROLES for role in user.roles):
+            confirmation_view = _utils.StaffBanConfirmationView(
+                allowed_users=(ctx.author.id,),
+                allowed_roles=constants.MODERATION_ROLES,
+                timeout=10,
+            )
+
+            await confirmation_view.send(
+                ctx.channel,
+                message_content="That user is a Helper. Are you sure you want to ban them?",
+            )
+
+            await confirmation_view.wait()
+            if confirmation_view.value is None:
+                log.trace(
+                    "Attempted ban of user %s by moderator %s cancelled due to timeout.",
+                    str(user),
+                    str(ctx.author),
+                )
+
+            elif confirmation_view.value is False:
+                log.trace(
+                    "Attempted ban of user %s by moderator %s cancelled due to manual cancel.",
+                    str(user),
+                    str(ctx.author),
+                )
+
+            return None
```

Comment:
Am I missing something? You're `returning` if the value is `True` aka even if the staff confirms the ban it will exit the function

Reply:
Right. Logical fallacy on my part. Will fix ASAP